webhooks/virtualmachinepublishrequest/validation/virtualmachinepublishrequest_validator_intg_test.go

```diff
@@ -62,6 +64,15 @@ func intgTestsValidateCreate() {
 
 		Expect(ctx.Client.Create(ctx, ctx.vm)).To(Succeed())
 		Expect(ctx.Client.Create(ctx, ctx.cl)).To(Succeed())
+		// Ensure VM and CL all exist before VMPub is created.
+		Eventually(func() error {
+			vmObj := &vmopv1.VirtualMachine{}
+			if err = ctx.Client.Get(ctx, ctrlclient.ObjectKeyFromObject(ctx.vm), vmObj); err != nil {
```

Comment:
Make sense. Updated to use a `eventually` block to check vmPub creation instead of checking resource existence before creation. This just wants to tackle the bug when target item name is missing. Add the eventually block is a workaround to fix the integration test race.
Yeah there is some duplicate logic in webhook and controller, we can handle this later. (For internal consistency, maybe just to focus on API/Kind check, as well as immutable fields check? But I do think use a validation webhook to gate creation can kind of improve UX?)